documentation/docs/assertions/json/schema.md

```diff
@@ -0,0 +1,92 @@
+---
+title: JSON Schema Matchers
+slug: json-schema-matchers.html
+sidebar_label: Schema matchers
+---
+
+| Matcher             | Description                                                                                                                                         | Targets       |
+|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------|:--------------|
+| `shouldMatchSchema` | Validates that a `String` or `kotlinx.serialization.JsonElement` matches a `JsonSchema`. See description below for details on constructing schemas. | Multiplatform |
+
+## Defining Schemas
+
+A subset of [JSON Schemas](https://json-schema.org/) can be defined either by parsing a textual schema. Example:
+
+```kotlin
+val parsedSchema = parseSchema(
+  """
+  {
+  "$id": "https://example.com/geographical-location.schema.json",  // will  be ignored
+  "$schema": "https://json-schema.org/draft/2020-12/schema",       // will be ignored
+  "title": "Longitude and Latitude Values",                        // will be ignored
+  "description": "A geographical coordinate.",                     // will be ignored
+  "required": [ "latitude", "longitude" ],
+  "type": "object",
+  "properties": {
+    "latitude": {
+      "type": "number",
+      "minimum": -90,
+      "maximum": 90
+    },
+    "longitude": {
+      "type": "number",
+      "minimum": -180,
+      "maximum": 180
+    }
+  }
+}
+  """
+)
+```
+
+or using Kotest's built-in DSL:
+
+```kotlin
+val addressSchema = jsonSchema {
+  obj {   // object is reserved, obj was chosen over jsonObject for brevity but could be changed ofc, or jsonObject could be added as alternative.
+    withProperty("street", required = true) { string() }
```

Comment:
What do you think about allowing `string("name", true)` as well, for simplicity ?

Reply:
Yeah, that would be neat :)

Reply:
Everything is marked as experimental. Would you be ok with dropping a 5.3.1 with some DSL-additions or wait a day or two for me to finish that?

Reply:
5.3.1 is fine, or later even.
Gives us chance to gain more feedback.